Tools/clinic/clinic.py

```diff
@@ -1730,15 +1736,15 @@ def __init__(self, input, language, *, verify=True):
         self.find_start_re = create_regex(before, after, whole_line=False)
         self.start_re = create_regex(before, after)
         self.verify = verify
-        self.last_checksum_re = None
-        self.last_dsl_name = None
-        self.dsl_name = None
+        self.last_checksum_re: re.Pattern[str] | None = None
+        self.last_dsl_name: str | None = None
+        self.dsl_name: str | None = None
         self.first_block = True
 
-    def __iter__(self):
+    def __iter__(self) -> BlockParser:
```

Comment:
IIUC, we'd be better off using `Self` here, but, since that was introduced in 3.11...

Reply:
Yes. We could use a TypeVar to achieve the same effect on <=3.10, but there's no real need — since we have no intention of subclassing this class, returning `BlockParser` here works just as well for our needs :)